askMenuOption.cpp

```diff
@@ -1,23 +1,34 @@
-#pragma once
-#include <iostream>
-#include <map>
-
-#include "askMenuOption.h"
-
-void AskMenuOption::execute() {
-	std::cout << "Choose option: ";
-}
-
-PossibleState AskMenuOption::leave() {
-	int inputOption;
-	std::cin >> inputOption;
-
-	PossibleState nextState;
-	std::map<int, PossibleState> nextStateMap{
-		{1, PossibleState::addTask},
-		{2, PossibleState::showTasks},
-		{3, PossibleState::editTask}
-	};
-
-	return nextState = nextStateMap.at(inputOption);
+#include<iostream>
+#include<map>
+
+#include"askMenuOption.h"
+
+void AskMenuOption::nextStateSelection() {
+	int inputOption;
+	int* input_ptr;
+	input_ptr = &inputOption;
+	std::cout << "Choose option: ";
+	std::cin >> inputOption;
+	std::map<int, PossibleState> nextStateMap{
```

Comment:
To mi się nie podoba ale ograniczenie wynika z architektury (moja wina).
Teraz można kłamać dowolnie w menu wypisując co użytkownik może kliknąć.
Najlepiej by było jakby menu sobie definiowało te mapę. Wypisywało co można kliknąć 
na jej podstawie.

Jest jedno obejście. Możesz te mapę zdefiniować w tej twojej klasie Config.
Wtedy unikniemy jednej złej rzeczy "tight couplingu" klas stanów.
Czyli sytuacji w której jeden stan potrzebowałby includeować drugi stan żeby pobrać jakieś informacje.
(między stanami a Config już i tak jest)